addon/mixins/application-route-mixin.js

```diff
@@ -6,6 +6,49 @@ import { getOwner } from '@ember/application';
 import { inject } from '@ember/service';
 import Ember from 'ember';
 import Configuration from './../configuration';
+import { assert } from '@ember/debug';
+import location from 'ember-simple-auth/utils/location';
+/**
+ * Standard after-session-validation logic
+ * @param {Ember.Route | Ember.Router} rou A container-aware ember object with a transitionTo() function
+ * @param {string} routeAfterAuthentication Route to redirect the user to, if no previous transition information is found
+ */
+export function sessionAuthenticatedHandler(rou, routeAfterAuthentication) {
```

Comment:
Can we name the param `router` instead?

Reply:
Maybe call the method `handleSessionAuthenticated`? I'd reserve nouns for classes etc.